packages/shr-fhir-export/lib/export.js

```diff
@@ -4591,6 +4595,41 @@ function isTargetBasedOn(target, baseTarget, targetSpec, configuration) {
   return false;
 }
 
+function typesSameAsOriginal(type) {
+  // For every type, check if the type is the same as the original
+  let allTypesMatch = true;
+  for (const t of type) {
+    const originalTargets = t._originalTargetProfiles;
+    const targets = MVH.typeTargetProfile(t);
+    const originalProfiles = t._originalProfiles;
+    const profiles = MVH.typeProfile(t);
+    let targetsMatch = false;
+    let profilesMatch = false;
+    if (originalTargets != null && targets != null) {
+      // Alphabetically sort to make comparison easy
+      const sortedOriginalTargets = [...originalTargets].sort();
+      const sortedCurrentTargets = [...targets].sort();
+      // See if the arrays are the same
+      targetsMatch = sortedOriginalTargets.every((el, i) => el === sortedCurrentTargets[i]);
+    } else if (originalTargets == null && targets == null) {
+      targetsMatch = true;
+    }
+
+    if (originalProfiles != null && profiles != null) {
+      // Alphabetically sort to make comparison easy
+      const sortedOriginalProfiles = [...originalProfiles].sort();
+      const sortedCurrentProfiles = [...profiles].sort();
+      // See if the arrays are the same
+      targetsMatch = sortedOriginalProfiles.every((el, i) => el === sortedCurrentProfiles[i]);
+    } else if (originalProfiles == null && profiles == null) {
+      profilesMatch = true;
+    }
+    // If allTypesMatch is ever set to false, we want it to stay false
+    allTypesMatch = allTypesMatch && profilesMatch && targetsMatch;
```

Comment:
Wouldn't it be more efficient just to immediately return the first time you get a `false`?  E.g.,
```
if (!targetsMatch || !profilesMatch) {
  return false;
}
```
Then you skip all remaining iterations.  And you don't need to track an `allTypesMatch` variable.

Given that this function will get called a _lot_, I think it makes sense to go ahead and make this optimization.

Reply:
Huh, what was I thinking when I did this the first time? Thanks for catching that. Fixed and rebased.